### PR TITLE
Track query engine size over time

### PR DESCRIPTION
**Pull request by aqrln**
https://prisma.github.io/prisma-engines/engines-size/

### CODE REVIEW

File: tools/update-engine-size/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+#[derive(Serialize)]
+struct DbEntry<'a> {
+    date_time: &'a str,
+    branch: &'a str,
+    commit: &'a str,
+    file: &'a str,
+    size_bytes: u64,
+}
+
+fn main() -> color_eyre::Result<()> {
```
Comment (aqrln) [inline tools/update-engine-size/src/main.rs:39]: probably remove it to save on compile time, `Box<dyn Error>` is more than enough for a simple program like this

File: nix/publish-engine-size.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    text = ''
+      set -euxo pipefail
+
+      if ! git diff --exit-code 1> /dev/null; then
+        : "The workspace is not clean. Please commit or reset, then try again".
+        exit 1
+      fi
+
+      CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+      CURRENT_COMMIT=$(git rev-parse --short HEAD)
```
Comment (aqrln) [inline nix/publish-engine-size.nix:29]: use full commit hash here instead

File: tools/update-engine-size/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,74 @@
+use std::{
```
Comment (aqrln) [inline tools/update-engine-size/src/main.rs:1]: Can be replaced with a simple shell script now, the initial idea was it was going to do more than this

File: nix/publish-engine-size.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        exit 1
+      fi
+
+      CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+      CURRENT_COMMIT_SHORT=$(git rev-parse --short HEAD)
+      CURRENT_COMMIT_FULL=$(git rev-parse HEAD)
+      REPO_ROOT=$(git rev-parse --show-toplevel)
+
+      pushd "$REPO_ROOT"
+      git fetch --depth=1 origin gh-pages
```
Comment (janpio) [inline nix/publish-engine-size.nix:34]: Let's highlight somehow that this writes into the `gh-pages` branch, maybe a documenation block how it works that explains it.

File: nix/publish-engine-size.nix [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+      if [[ ! -f $CSV_PATH ]]; then
+        echo "date_time,branch,commit,file,size_bytes" > "$CSV_PATH"
+      fi
+
+      for file in "$@"; do
+        file_name=$(basename "$file")
+        size=$(stat -c %s "$file")
+        echo "$DATE_TIME,$CURRENT_BRANCH,$CURRENT_COMMIT,$file_name,$size" >> "$CSV_PATH"
+      done
```
Comment (tomhoule) [inline nix/publish-engine-size.nix:73]: :chefkiss:

**Review by tomhoule — APPROVED** [review]
Clean code and a very good first version, awesome job :100: 

Small improvements on the side we can make later will be the TODO you noted and merging the engines size github actions and the cargo docs github action, so we don't need to set up the nix store, check out the gh pages branch, etc. twice.

**Comment by tomhoule** [comment]
closes https://github.com/prisma/client-planning/issues/178

**Pull request merged**